internal/pkg/cli/deploy/env.go

```diff
@@ -100,11 +141,115 @@ func NewEnvDeployer(in *NewEnvDeployerInput) (*envDeployer, error) {
 			Env:             in.Env,
 		},
 		newStackSerializer: func(in *deploy.CreateEnvironmentInput, lastForceUpdateID string, oldParams []*awscfn.Parameter) stackSerializer {
-			return stack.NewEnvConfigFromExistingStack(in, lastForceUpdateID, oldParams)
+			return cfnstack.NewEnvConfigFromExistingStack(in, lastForceUpdateID, oldParams)
+		},
+		envDescriber:      envDescriber,
+		envManagerSession: envManagerSession,
+		newLBDescriber: func(sess *session.Session) lbDescriber {
+			return elbv2.New(sess)
+		},
+		newServiceStackDescriber: func(svc string, sess *session.Session) stackDescriber {
+			return stack.NewStackDescriber(cfnstack.NameForService(in.App.Name, in.Env.Name, svc), sess)
 		},
 	}, nil
 }
 
+// Verify checks that the manifest configuration is valid to deploy alongside
+// the currently deployed resources.
+func (d *envDeployer) Verify(ctx context.Context, mft *manifest.Environment) error {
```

Comment:
What do you think of renaming the method to `Validate` to be consistent with our other packages (like `manifest`)?
```suggestion
// Validate returns an error if the environment manifest is incompatible with services and application configurations.
func (d *envDeployer) Validate(ctx context.Context, mft *manifest.Environment) error {
```